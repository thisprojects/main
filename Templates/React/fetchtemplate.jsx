import React from 'react';
import Fetch from 'react-fetch';






class Welcome extends React.Component {
    render (){
        return (<h1>Hello1 {this.props.value}</h1>);


    }

}

class List extends React.Component {

                render () {
   return (
       <ul>
           {this.props.list.map(station => <li key={station}> {station} </li>)}
       </ul>
   );
}



}


class ShoppingList extends React.Component {

constructor (props){
            super(props);
            this.state = {value: "hello", schnups: false, list: ["one","two","three"], results: [] }

        }

componentDidMount () {

 this._fetchIT ();

}


  render() {



    return (
      <div className="shopping-list">
        <h1>List of Stuff</h1>

        <ul>
        <List list={this.state.list}/>
        </ul>
        <Welcome value={this.state.value}/>
        {this._doStuff()}
        {this._jsonList()}
        <div>

        </div>
        <button onClick={this._testSt.bind(this, this.state.schnups)}>Click me!</button>
      </div>

    );
  }

_doStuff () {

    if (this.state.schnups){

    return (<h1> Bibobs </h1>);
     }
}

_testSt () {
        this.setState({value: "wibbums", schnups: !this.state.schnups, list: ["one", "two", "three", "four", 13] });
        this.setState({results: []});
        this._fetchIT();

    }
_jsonList (){

    if (this.state.results.length){
        let starwars = this.state.results
        return (
            <div>

                {starwars.map (starwars  =>
                <ul>
                <li key={starwars.toString()}>{starwars.name}</li>
                </ul>
                )}
            </div>
        )


    }else{

        return (

            <div>
            <h1>Loading!...</h1>
            </div>
        )
    }


}

_fetchIT () {
    fetch('https://swapi.co/api/people/')
   .then(response => response.json())
      .then(data => this.setState({ results: data.results }));


}

}

export default ShoppingList;
