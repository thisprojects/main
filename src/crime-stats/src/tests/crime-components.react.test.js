import React from 'react';
import renderer from 'react-test-renderer';
import {
  MapCrimeResults,
  SortCrimes,
  CrimeStats,
} from '../main-components/crime-components.js';
import {
  mockResults
} from '../tests/mocked-results.js';


test('Map Crime Results Matches Snapshot', () => {
  const component = renderer.create(
    <MapCrimeResults results={mockResults} applyLocationSort={()=>{}} applyOutcomeSort={()=>{}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Sort Crimes Matches Snapshot', () => {
  const component = renderer.create(
    <SortCrimes totalCrimes={"1337"} locationSort={()=>{}} outcomeSort={()=>{}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Crime Stats Matches Snapshot', () => {
  const component = renderer.create(
    <CrimeStats results={mockResults} applyLocationSort={()=>{}} applyOutcomeSort={()=>{}} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});