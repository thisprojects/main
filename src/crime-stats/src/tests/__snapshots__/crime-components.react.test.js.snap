// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crime Stats Matches Snapshot 1`] = `
<div
  className="crime-stats"
>
  <div
    className="total-crimes"
  >
    <p>
      Total Crimes 
      3
    </p>
    <input
      name="sort"
      onChange={[Function]}
      type="radio"
      value="location"
    />
    <p>
      Location Sort
    </p>
    <input
      name="sort"
      onChange={[Function]}
      type="radio"
      value="outcome"
    />
    <p>
      Outcome Sort
    </p>
  </div>
  <div
    className="results-table"
  >
    <div
      className="stat-row"
    >
      <span
        className="result-category"
      >
        ccc Mock Category 1
      </span>
      <span
        className="result-location"
      >
        aaa Mock Street 1
      </span>
      <span
        className="result-outcome"
      >
        ccc Mock Outcome 1
      </span>
    </div>
    <div
      className="stat-row"
    >
      <span
        className="result-category"
      >
        bbb Mock Category 2
      </span>
      <span
        className="result-location"
      >
        ccc Mock Street 2
      </span>
      <span
        className="result-outcome"
      >
        bbb Mock Outcome 2
      </span>
    </div>
    <div
      className="stat-row"
    >
      <span
        className="result-category"
      >
        aaa Mock Category 3
      </span>
      <span
        className="result-location"
      >
        bbb Mock Street 3
      </span>
      <span
        className="result-outcome"
      >
        aaa Mock Outcome 3
      </span>
    </div>
  </div>
</div>
`;

exports[`Map Crime Results Matches Snapshot 1`] = `
Array [
  <div
    className="stat-row"
  >
    <span
      className="result-category"
    >
      ccc Mock Category 1
    </span>
    <span
      className="result-location"
    >
      aaa Mock Street 1
    </span>
    <span
      className="result-outcome"
    >
      ccc Mock Outcome 1
    </span>
  </div>,
  <div
    className="stat-row"
  >
    <span
      className="result-category"
    >
      bbb Mock Category 2
    </span>
    <span
      className="result-location"
    >
      ccc Mock Street 2
    </span>
    <span
      className="result-outcome"
    >
      bbb Mock Outcome 2
    </span>
  </div>,
  <div
    className="stat-row"
  >
    <span
      className="result-category"
    >
      aaa Mock Category 3
    </span>
    <span
      className="result-location"
    >
      bbb Mock Street 3
    </span>
    <span
      className="result-outcome"
    >
      aaa Mock Outcome 3
    </span>
  </div>,
]
`;

exports[`Sort Crimes Matches Snapshot 1`] = `
<div
  className="total-crimes"
>
  <p>
    Total Crimes 
    1337
  </p>
  <input
    name="sort"
    onChange={[Function]}
    type="radio"
    value="location"
  />
  <p>
    Location Sort
  </p>
  <input
    name="sort"
    onChange={[Function]}
    type="radio"
    value="outcome"
  />
  <p>
    Outcome Sort
  </p>
</div>
`;
